### PR TITLE
[JENKINS-4433] Upgrade log4j to 1.2.17

### PR DESCRIPTION
**Pull request by eidottermihi**

### CODE REVIEW

File: core/pom.xml [hunk 1]
```diff
@@ -468,6 +468,11 @@ THE SOFTWARE.
       <artifactId>commons-logging</artifactId>
     </dependency>
     <dependency>
+      <groupId>log4j</groupId>
+      <artifactId>log4j</artifactId>
+      <version>1.2.17</version>
```
Comment (KostyaSha) [inline core/pom.xml]: Where it initially defined? Shouldn't be enough to set version in properties?

Comment (eidottermihi) [inline core/pom.xml]: It was not explicitily defined, but rather pulled in as transitive dependency of acegi-security ([see JENKINS-4433] (https://issues.jenkins-ci.org/browse/JENKINS-4433?focusedCommentId=229977&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-229977)).

**Comment by KostyaSha** [comment]
@uhafner please next time wait for :+1: from more then one person.
@jglick highlight

File: core/pom.xml (hunk 1 above)
Comment (KostyaSha) [inline core/pom.xml]: Prefered way is define version in properties section, but @jglick comments/review will be welcome.

Comment (uhafner) [inline core/pom.xml]: Hmm, most other dependencies don't use a property, so shouldn't this change be consistent with the existing configuration?

**Comment by uhafner** [comment]
@KostyaSha do we have a new code reviewing guideline?

File: core/pom.xml (hunk 1 above)
Comment (KostyaSha) [inline core/pom.xml]: Most dependencies using properties, press plus in diff before this change. Also you can navigate upto https://github.com/jenkinsci/pom/blob/master/pom.xml#L611 but it ancient and variables mostly overrided in jenkins poms.

Comment (KostyaSha) [inline core/pom.xml]: This also allows plugins overriding versions without overriding xml sections.

Comment (uhafner) [inline core/pom.xml]: These properties are for maven plug-ins, not for libraries. Most properties of dependencies are hard-coded. (But anyway, I also think that properties are easier to change. My point is that we should then do this for *all* and not just for a subset of these).

Comment (KostyaSha) [inline core/pom.xml]: > These properties are for maven plug-ins, not for libraries.

```
 <properties>
    <staplerFork>true</staplerFork>
    <stapler.version>1.237</stapler.version>
    <spring.version>2.5.6.SEC03</spring.version>
    <groovy.version>1.8.9</groovy.version>
  </properties>
```

**Comment by KostyaSha** [comment]
> @KostyaSha do we have a new code reviewing guideline?

It used in all other PRs.

**Comment by jtnord** [comment]
we shouldn't be shipping log4j at all.

Log4j needs to be rewritten to JUL (as does slf4j) so log4j should be excluded (it was) and log4j-over-slf4j should really be used instead.

A belated :-1:

**Comment by uhafner** [comment]
But we are already shipping it, log4j is visible to all plugins. So this PR does not alter this state: it just makes this dependency visible and uses a version that is compatible to several plugins, see JENKINS-4433.

**Comment by jtnord** [comment]
@uhafner that we are already shipping it is  surely be a bug - Im's sure it was not meant to be the case and has brought in accidentally by a transitive dependency - hence the existing exclusion to prevent that from happening..  If we ship log4j and people use it we have multiple logging systems which cannot be controlled.

**Comment by daniel-beck** [comment]
@jtnord @KostyaSha So should this change be reverted?

**Comment by jtnord** [comment]
@daniel-beck I believe we should not be shipping log4j but use the slf4j log4j bridge so that any calls to log4j get written via java.util.logging which is the Jenkins logging way.

Reverting this fix won't actually get that - as it appears as though something has transitively included log4j.  So the root-cause needs to be addressed

**Pull request merged**